frontend/src/components/pagedItemsTable/PagedItemsTable.tsx

```diff
@@ -133,18 +135,76 @@ function PagedItemsTable<T extends object>({
 		[sortBy, setSortByCallback]
 	)
 
-	// If a selection listener is passed to the table then the listener is informed
-	// every time the row selection is changed by the user.
-	let rowSelection: TableRowSelection<T> | undefined = undefined
-	if (selection) {
-		rowSelection = {
-			type: 'checkbox',
-			onChange: (selectedRowKeys: React.Key[], selectedRows: T[]) => {
-				selection.onSelectionChanged(selectedRows)
-			},
-			selectedRowKeys: [...selection.selectedItemIDs],
+	// Return the ID that corresponds to the object displayed in a row of the table.
+	// We just find the object in the dataObjects map and return its corresponding
+	// key. This allows us to use data objects that don't have an explicit 'id' property.
+	const UNKOWN_DATA_ROW_KEY = 'unknown-data-row-key'
+	const getRowKeyForDataObject = useCallback((data: T) => {
+		for (const key in tableDataState.objectMap) {
+			if (tableDataState.objectMap[key] === data) {
+				return key
+			}
 		}
-	}
+		return UNKOWN_DATA_ROW_KEY
```

Comment:
Is it possible to have a table row without key ? Is it the best behaviour to create a false key that could mimic as a real key. From my understanding, it could lead to a selection silently missing items ... The missing keys are ignored when listing the selection. If it is not possible, this code should not be there, if it is possible it should not behave that way because we want this to be obvious there is something fishy.

Reply:
> if it is possible it should not behave that way because we want this to be obvious there is something fishy
Throw exception?

Reply:
```ts
	const onSelectSingle = useCallback((record: T) => {
		const key = getRowKeyForDataObject(record)
		if (key !== UNKOWN_DATA_ROW_KEY) {
			let newSelectedItems = selectedItems
			if (selectedItems.includes(key)) {
				newSelectedItems = selectedItems.filter((id) => id !== key)
			} else {
				newSelectedItems = [...selectedItems, key]
			}
			setSelectedItems(newSelectedItems)
			if (selection) {
				selection.onSelectionChanged(newSelectedItems, selectAll)
			}
		}
	}, [getRowKeyForDataObject, selectAll, selectedItems, selection])

```

Reply:
throw notification i would say in case there is no valid key i would notify the user the selection failed for some unknown reason ... To me this looks like coding for bad coding ... I don't see why we would have a table with no key and use this function to look for a key ... I personally would just remove the default key and if the key is undefined fail more or less gracefully but obviously so we can fix that case.